### DISCUSSION TITLE
PAL PS1 game output at 60hz instead of 50hz in Default Video Mode

### DISCUSSION THREAD

**Discussion by Pedrofsky** [aldostools/IRISMAN#59] [Q&A] [answered]
Hello.

I'm using IRISMAN 4.89 on my PAL CFW PS3 (Evilnat 4.89) to run my PS1 backup games.

I've encountered an issue with my PAL PS1 games when running upscaled (IRISMAN Game Option Video Mode set to Default).

When I run a PAL PS1 backup game through IRISMAN, video output is 720p at 60 hz ; which causes sound to play faster than intended (image is probably faster too but it's less noticeable). I've tested Duke Nukem for example, and on the main menu the music plays too fast ; and Duke's voice sounds a bit like a little girl.

I own the original disc and I when play the disc directly through XMB, the PS3 outputs 720p at 50 Hz. Moreover, if I play an NTSC disc (like US version of Point Blank that I own) directly through XMB, the PS3 outputs correctly at 60 Hz. It seems that the PS3 is able to identify and apply the correct refresh rate for a given PS1 disc...

...Where IRISMAN seems to always output at 60 Hz regardless of the disc region, in Default Video Mode. If I set 576 as Video Mode in IRISMAN per-game settings, the refresh rate is 50 Hz. But then I lose the benefit of upscaling... And if I play directly though XMB, I lose the benefits of the ps1_netemu emulator... Which really makes a difference in certain games (Duke Nukem is one of them, with many visual glitches cleaned up thanks to ps1_netemu).

Is there something I'm missing here ?

Also, I don't know if it has anything to do with the issue, but I noticed that IRISMAN outputs at 480p my PAL PS1 games when Video Mode is set to Region Disc... which sounds like an error to me. If IRISMAN read correctly the Disc Region of my PAL games, it would output 576p and not 480p. So I'm wondering if it's not related... But this, I do not care much, as I can always set 480 or 576 manually if I intend to play in low resolution.

I also noticed that if I set the emulator to ps1_emu, while IRISMAN outputs at 720p at 60hz, the sound actually plays at normal speed. But visually the game plays worse (as expected with many games with ps1_emu compared to ps1_netemu).

I also tested running my PAL Duke Nukem backup through XMB mounting it via WebmanMOD (and forcing ps1_netemu by holding R2 button) : it works as intended, 720p at 50 hz. So I definitely feel like it's an issue with IRISMAN. It's a shame because I love how simple and clean IRISMAN looks and feels compared to WebmanMOD.

EDIT : to make it more simple, here are my observations :

- I launch my PSX games with IRISMAN only in Direct mode
- I have trouble with the sound playing too fast whenever I play a PAL game with ps1_netemu emulator and Video Mode set to Default
- I notice the Region Disc Video Mode triggers a 480p/60hz resolution instead of 576p/50hz which should be the one for PAL games (but the sound plays fine)
- I notice that both ps1_emu and ps1_netemu in Default Video Mode (with PS3 resolution set at 720p) output at 720p/60hz but the sound is playing too fast only with ps1_netemu
- Launching the same games via XMB after mounting them with Webman MOD and forcing ps1_netemu with R2 button results in the sound playing at normal speed and video outputting at 720p/50hz

**Comment by aldostools** [comment]
IRISMAN has 2 modes of launching PSX games:
- Direct mode: the game is launched using a PSX payload by Estwald. It is used if the game is stored on internal HDD or FAT32.
- Cobra mode: the game is mounted and IRISMAN exits to XMB. The game is controlled by Cobra/Mamba payload. This mode is used when the game is stored in a NTFS disk.

For the second mode, latest MAMBA allows to use the tags "NTSC" or "PAL" in the file name of the ISO to force the region/refresh rate.
In Cobra, the feature is pending to be implemented among other features like LibCrypt subchannel data support.

For the direct mode, IRISMAN reads the sector 4 of the ISO to identify the region from the the license message.
While Cobra uses the game id in the executable to detect the region.

Before make some change in the code, we need to know how the game is being loaded and if happens the same using other backup managers like multiMAN or webMAN MOD.

**Comment by Pedrofsky** [comment]
All my games are stored either on internal HDD or on a FAT32 USB drive : therefore I deduce I exclusively use Direct mode.

Regarding other backup managers : I edited my first message to add the following :

_I also tested running my PAL Duke Nukem backup through XMB mounting it via WebmanMOD (and forcing ps1_netemu by holding R2 button) : it works as intended, 720p at 50 hz._

Is it enough information or do you need anything else ?

**Reply by Pedrofsky** [reply]
Here are the links to download the PAL Duke Nukem backup I use :

https://videogiciels.fr/public/DUKENUKEM.BIN
https://videogiciels.fr/public/DUKENUKEM.CUE

(uploaded on my personal website)

**Comment by aldostools** [comment]
Thank you for the information about the game.
I will check it after work.

**Reply by Pedrofsky** [reply]
I summarized my observations in an edit in my original post.

**Comment by aldostools** [comment]
I reviewed the game and effectively the game is running using the default settings based on the current video mode selected (e,g, 1080p or 720p).

If you press TRIANGLE on the game, then go to the option Video / Others, press X, in the first option Video Mode select Region Disc or 576. That will change the refresh rate from 60hz to 50hz (PAL).

**Reply by Pedrofsky** [reply]
Okay, but then I'm losing the benefit of upscaling. I would like to play at 720p with the sound playing at normal speed.

**Comment by aldostools** [comment]
In that case you need to press SELECT+X to skip the direct mode and launch the game mounted from XMB.

The payload for direct mode in IRISMAN doesn't have an option to specify the refresh rate other than select 576 for 50hz.

**Reply by Pedrofsky** [reply]
If I press SELECT+X to then launch the game mounted from XMB, in-game I encounter visual glitches due to ps1_emu being used. Is there a command to force ps1_netemu instead ?

**Comment by aldostools** [comment]
If you use the direct mode, there is an option to select the emulator in the TRIANGLE menu.
If you skip the direct mode, the default ps1 emulator is used.

If you use webMAN MOD you can create a custom combo to swap the ps1 emulators or if you mount the game with webMAN MOD you can select the ps1 emulator pressing R2 for ps1_netemu or L2 for ps1_emu when you mount the game.

NOTE: All map paths are cleared when you start IRISMAN or multiMAN, so you have to swap the emulator after launch the backup manager.

**Reply by Pedrofsky** [reply]
Understood. But just to be sure... I'm sorry to insist and I thank you for your availability, it is greatly appreciated. My real problem is sound playback speed. Are you sure it has to do with the refresh rate ? I know I pushed this explanation, but as I mentioned earlier, when playing in 480p/60hz the sound plays fine... So I'm wondering if there could be something else that could be tweaked in the code to ensure the sound plays at normal speed in upscaled PAL games ran with ps1_netemu ?

**Comment by aldostools** [comment, marked as answer]
When you play the game from XMB using Cobra or MAMBA, the payload has a feature that switches the region internally.
That's why the game sounds fine using 720p or 1080p or 480p.

When the game is launched by IRISMAN using the direct mode, a different payload is used. It executes the emulator directly, bypassing all the checks in Cobra/Mamba.

If you want to skip the direct mode, you need to use SELECT+X

**Comment by Pedrofsky** [comment]
Hi @aldostools ,

By any chance, has the situation evolved since august 2022 ?

I mean : is IRISMAN Direct Mode now able to run PS1 games with ps1_netemu at 720p _and_ 50 hz ?

**Comment by aldostools** [comment]
There haven't been much changes in IRISMAN in the recent years since my development is mainly focused on webMAN MOD.

However, PS1 emulators have been updated in the final version of Evilnat 4.91.2. You can check it solves your issue.

As far as I remember, in Cobra mode PS1 games play fine at 720p @ 50 hz using recent Cobra/Mamba/PS3HEN payloads thanks to a recent feature that lets you force the PS1 game to play in "NTSC" or "PAL" mode if these strings are included in the file name of the ISO.

Also another module patch added lately to the payload forces the product code 0x85 for PAL games, whereas NTSC gmes use 0x82 (debug).

If I remember correctly, IRISMAN lets mount the PS1 games in FAT32 and the internal HDD with Cobra payload pressing SELECT.
Otherwise the direct launch mode will be used, except for the games stored in NTFS that also will use the Cobra payload.

These patches are implemented in the latest releases of Cobra/Mamba/PS3HEN.
